### PR TITLE
[FLINK-20058][kafka connector] Improve tests for per-partition-waterm…

### PR DESCRIPTION
**Pull request by fsk119**
…ark kafka

## What is the purpose of the change

*Improve the tests for per-partition-watermark kafka.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by fsk119** [comment]
@flinkbot run azure

File: flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesRuntimeFunctions.java [hunk 1]
```diff
@@ -318,9 +318,9 @@ public void invoke(RowData value, Context context) throws Exception {
 				Row row = (Row) converter.toExternal(value);
 				assert row != null;
 				if (rowtimeIndex >= 0) {
-					LocalDateTime rowtime = (LocalDateTime) row.getField(rowtimeIndex);
+					TimestampData rowtime = TimestampData.fromLocalDateTime((LocalDateTime) row.getField(rowtimeIndex));
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesRuntimeFunctions.java:321]:
```java
					// currently, rowtime attribute always using 3 precision
					TimestampData rowtime = value.getTimestamp(rowtimeIndex, 3);
```

File: flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesRuntimeFunctions.java [hunk 2]
```diff
@@ -318,9 +318,9 @@ public void invoke(RowData value, Context context) throws Exception {
 				Row row = (Row) converter.toExternal(value);
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesRuntimeFunctions.java:318]: Remove the `@SuppressWarnings("rawtypes")` annotation on this method.

**Review by wuchong — APPROVED** [review]
LGTM.

**Comment by fsk119** [comment]
@flinkbot run azure

**Comment by wuchong** [comment]
Will merge this, the failed test is not related to the pull request which has been tracked in FLINK-20065.

**Pull request merged**